### PR TITLE
Fixed a bug with image warnings in the browser 

### PR DESCRIPTION
**Pull request by IhorMasechko**
Fixed a bug with image warnings in the browser console on the homepage, shop page, shop/id, and delivery and payment pages

## Summary by CodeRabbit

- **New Features**
	- Enhanced responsive design for product card, payment options, and image gallery components.

- **Bug Fixes**
	- Improved hover effects for better visual feedback on product cards.

- **Style**
	- Adjusted image and text sizes for better readability on smaller screens.
	- Ensured images within payment options maintain a maximum width of 100%.
	- Introduced dynamic height adjustments for gallery images on smaller screens, maintaining aspect ratios.
	- Refined hover states and visibility of color options in product cards for improved user interaction.

### CODE REVIEW

**Review by VitalyyP — DISMISSED** [review]
LGTM

File: src/components/common/productCard/productCard.module.css [hunk 1]
```diff
@@ -139,6 +139,7 @@
   }
 
   .imgWrapper img {
-    width: 343px;
+    width: 100%;
```
Comment (yuramax) [inline src/components/common/productCard/productCard.module.css:142]: @IhorMasechko, if you set the width to 100%, the image size for the mobile version will be incorrect compared to how it is drawn in the design in Figma, see the attached screenshot. You can use both `max-width` and `min-width` property for this case. Please contact me if you need any clarification.

wrong-product-image-size-mobile

File: src/components/deliveryPaymentPage/deliveryAndPayment/DeliveryAndPayment.module.css [hunk 2]
```diff
@@ -51,12 +51,10 @@
...
 }
 
-.paymentOption img {
-  max-width: 100%;
-  height: auto;
-}
-
 @media (max-width: 1023px) {
+  .paymentOption img {
+    max-width: 100%;
```
Comment (yuramax) [inline src/components/deliveryPaymentPage/deliveryAndPayment/DeliveryAndPayment.module.css:56]: I still get warnings on Delivery and Payment page for mobile version.

delivery-and-payment-image-warnings

**Review by yuramax — CHANGES_REQUESTED** [review]
It's not easy to fix, so don't hesitate to contact me for help - I'll be happy to show you a few tricks 😉

File: src/components/common/productCard/productCard.module.css (hunk 1 above)
Comment (yuramax) [reply src/components/common/productCard/productCard.module.css]: I have reviewed the fixes and it looks OK now! Good job!

File: src/components/deliveryPaymentPage/deliveryAndPayment/DeliveryAndPayment.module.css (hunk 2 above)
Comment (yuramax) [reply src/components/deliveryPaymentPage/deliveryAndPayment/DeliveryAndPayment.module.css]: The changes works for me! 👍  Thanks.

**Review by yuramax — APPROVED** [review]
LGTM! 👌

**Review by VitalyyP — APPROVED** [review]
LGTM

**Pull request merged**